ksql-engine/src/test/resources/query-validation-tests/between.json

```diff
@@ -0,0 +1,147 @@
+        {"topic": "test_topic", "key": 3, "value": {"source": 2.1}, "timestamp": 0},
+        {"topic": "test_topic", "key": 4, "value": {"source": 3.9}, "timestamp": 0},
+        {"topic": "test_topic", "key": 5, "value": {"source": 4.0}, "timestamp": 0}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": 2, "value": {"THING": 2.0}, "timestamp": 0},
+        {"topic": "OUTPUT", "key": 3, "value": {"THING": 2.1}, "timestamp": 0},
+        {"topic": "OUTPUT", "key": 4, "value": {"THING": 3.9}, "timestamp": 0}
+      ]
+    },
+    {
+      "name": "test BETWEEN with string values",
+      "format": ["JSON"],
+      "statements": [
+        "CREATE STREAM TEST (source string) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT source AS THING FROM TEST WHERE source BETWEEN 'b' AND 'c';"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "key": 0, "value": {"source": null}, "timestamp": 0},
+        {"topic": "test_topic", "key": 1, "value": {"source": "a"}, "timestamp": 0},
+        {"topic": "test_topic", "key": 2, "value": {"source": "b"}, "timestamp": 0},
+        {"topic": "test_topic", "key": 3, "value": {"source": "ba"}, "timestamp": 0},
+        {"topic": "test_topic", "key": 4, "value": {"source": "c"}, "timestamp": 0},
+        {"topic": "test_topic", "key": 5, "value": {"source": "ca"}, "timestamp": 0}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": 2, "value": {"THING": "b"}, "timestamp": 0},
+        {"topic": "OUTPUT", "key": 3, "value": {"THING": "ba"}, "timestamp": 0},
+        {"topic": "OUTPUT", "key": 4, "value": {"THING": "c"}, "timestamp": 0}
+      ]
+    },
+    {
+      "name": "test NOT BETWEEN with integers",
+      "format": ["JSON"],
+      "statements": [
+        "CREATE STREAM TEST (source int) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT source AS THING FROM TEST WHERE source NOT BETWEEN 2 AND 4;"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "key": 1, "value": {"source": 1}, "timestamp": 0},
+        {"topic": "test_topic", "key": 2, "value": {"source": 2}, "timestamp": 0},
+        {"topic": "test_topic", "key": 3, "value": {"source": 3}, "timestamp": 0},
+        {"topic": "test_topic", "key": 4, "value": {"source": 4}, "timestamp": 0},
+        {"topic": "test_topic", "key": 5, "value": {"source": 5}, "timestamp": 0}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": 1, "value": {"THING": 1}, "timestamp": 0},
+        {"topic": "OUTPUT", "key": 5, "value": {"THING": 5}, "timestamp": 0}
+      ]
+    },
+    {
+      "name": "test BETWEEN with integers and variable values",
+      "format": ["JSON"],
+      "statements": [
+        "CREATE STREAM TEST (source int, min int, max int) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT source AS THING FROM TEST WHERE source BETWEEN min AND max;"
```

Comment:
Maybe add a test with expressions too. For instance, `source*2 BETWEEN min+2 AND max-min` or something similar.